packages/mui-codemod/src/v5.0.0/base-remove-component-prop.test/actual.tsx

```diff
@@ -0,0 +1,30 @@
+// @ts-nocheck
+import MaterialInput from '@mui/material/Input';
+import Input from '@mui/base/Input';
+import Switch from '@mui/base/Switch';
+import Badge from '@mui/base/Badge';
+
+<MaterialInput component={CustomRoot} />;
+
+<Input component={CustomRoot} />;
+
+<Input component={CustomRoot}></Input>;
+
+<Switch
+  component={CustomRoot}
+  randomProp="1"
+  randomProp2="2"
+  randomProp3="3"
+  slotProps={{ root: { className: 'root' } }}
+/>;
+
+<Badge
+  component={CustomRoot}
+  randomProp="1"
+  randomProp2="2"
+  randomProp3="3"
+  slots={{ badge: CustomBadge }}
+  slotProps={{ badge: { className: 'badge' } }}
+/>;
```

Comment:
```suggestion
<Badge
  slots={{ badge: CustomBadge }}
  component={CustomRoot}
  randomProp="1"
  randomProp2="2"
  randomProp3="3"
  slotProps={{ badge: { className: 'badge' } }}
/>;
```

To cover use-cases where the `slots` object is defined first.